query/src/main/java/org/infinispan/query/dsl/embedded/impl/SearchQueryBuilder.java

```diff
@@ -136,6 +138,7 @@ private <T> LuceneSearchQuery<T> build(SearchProjection<T> searchProjection) {
          queryOptionsStep = queryOptionsStep.failAfter(timeout, timeUnit);
       }
       queryOptionsStep = queryOptionsStep.totalHitCountThreshold(hitCountAccuracy);
+      queryOptionsStep.loading(l -> l.withMetadata(withMetadata));
```

Comment:
I don't know enough of HS, but maybe it is helpful to do this instead?

```java
      if (withMetadata) {
         queryOptionsStep.loading(l -> l.withMetadata(true));
      }
```

Reply:
The idea is to tell HibernateSearch to fetch the metadata only if need them. So I would keep it

Reply:
This would still do that. This is just us not calling the loading method if we don't need metadata.